app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/TrackerBlockingVpnService.kt

```diff
@@ -237,40 +216,29 @@ class TrackerBlockingVpnService : VpnService(), CoroutineScope by MainScope(), V
             ACTION_START_VPN, ACTION_ALWAYS_ON_START -> {
                 notifyVpnStart()
                 synchronized(this) {
-                    launch(serviceDispatcher) { startVpn() }
+                    launch(serviceDispatcher) {
+                        async {
+                            startVpn()
+                        }.await()
+                    }
```

Comment:
Add async/await to ensure that the subsequent coroutines will wait until startVpn() finishes

Reply:
nice!